src/editor/helpers.ts

```diff
@@ -170,42 +172,77 @@ export const isBlockActive = (editor: Editor, blockType: BlockElementType) => {
 export const toggleBlock = (editor: Editor, blockType: BlockElementType) => {
   const isActive = isBlockActive(editor, blockType);
 
-  // first, unwrap any lists
-  if (isListActive(editor)) {
-    Transforms.unwrapNodes(editor, {
-      split: true,
-      match: (elem) => isListType((elem as SluglineElement).type),
-    });
-    Transforms.setNodes(
-      editor,
-      { type: BlockElementType.Default },
-      {
-        match: (elem) =>
-          (elem as SluglineElement).type === BlockElementType.ListItem,
-      }
-    );
-  }
+  Editor.withoutNormalizing(editor, () => {
+    // first, unwrap any lists
+    if (isListActive(editor)) {
+      Transforms.unwrapNodes(editor, {
+        split: true,
+        match: (elem) => isListType((elem as SluglineElement).type),
+      });
+      Transforms.setNodes(
+        editor,
+        { type: BlockElementType.Default },
+        {
+          match: (elem) =>
+            (elem as SluglineElement).type === BlockElementType.ListItem,
+        }
+      );
+    }
 
-  // if the requested block is a list and that list is currently not active,
-  // we are toggling a list ON, and we have to do some wrapping
-  if (isListType(blockType) && !isActive) {
-    Transforms.setNodes(editor, { type: BlockElementType.ListItem });
-    Transforms.wrapNodes(editor, { type: blockType, children: [] });
-  }
-  // otherwise, toggle as usual
-  else if (isActive) {
-    Transforms.setNodes(
-      editor,
-      { type: BlockElementType.Default },
-      { mode: "all", match: (node) => Editor.isBlock(editor, node) }
-    );
-  } else {
-    Transforms.setNodes(
-      editor,
-      { type: blockType },
-      { mode: "all", match: (node) => Editor.isBlock(editor, node) }
-    );
-  }
+    // if the requested block is a list and that list is currently not active,
+    // we are toggling a list ON, and we have to do some wrapping
+    if (isListType(blockType) && !isActive) {
+      wrapListItems(editor, blockType);
+    }
+    // otherwise, toggle as usual
```

Comment:
nit: probably should move the comment within the else-if block for style consistency

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/84/commits/47f43914bb06fc8a929d8d02739d83683829f688